guides/developer/service-overviews/orca.md

```diff
@@ -0,0 +1,48 @@
+
+It's important to note that a Stage is recursively composable. One stage can have zero to many **Synthetic Stages**, that is, stages that can occur either `BEFORE` or `AFTER` its parent stage. This is most easily illustrated in a Canary stage, which deploys multiple Server Groups (a baseline and canary). While to the end-user this appears to be a single stage, it's actually a composition of multiple Deploy stages with some extra logic on top.
+
+Lastly, an Execution's stage graph does not need to be known ahead of time. An Execution can lay its tracks down ahead of itself as it is running, which is how some of the more advanced functionality is implemented, like automatic rollbacks or canary behaviors.
+
+## Runtime
+
+Orca uses a distributed queue library, [Keiko](http://github.com/spinnaker/keiko), to manage its work. Keiko uses atomic messages to progress work through its lifecycle and allows Orca to first be resilient to node failure, as well as spread load evenly across the deployment making Orca easier to scale and operate.
+
+Keiko itself is an abstraction around a delay queue, meaning it can deliver **Messages** immediately or at specific times in the future, as well as reschedule messages that have already been added to the queue for a new delivery time. This delay queue functionality is pivotal to Orca's performance and operational characteristics when dealing with long running operations.
+
+A Message is a fine-grained unit of work. As an example, a `StartStage` message will verify that all upstream branches are complete and then perform any logic associated with preparing for the stage's execution, it won't actually perform the stage's core actions. A `RunTask` is really the only message that actually performs stage work that a user would be familiar with.
+
+Some message types are redeliverable and duplicated, whereas others are not. For example, in a pipeline that has two concurrent stage branches that do not join at the end, there will be two `CompleteExecution` messages sent. This is fine, because the `CompleteExecutionHandler` has logic to verify that indeed all stages are in a completed state before marking the entire execution as complete, whether it ultimately be a failure or success.
+
+For every message, there is a single correlated `Handler` that contains all of the logic for processing this `Message`. A Message's contents are small and the bare minimum of information to process the request: For instance, an Execution is referenced by ID rather than containing its entire state. Orca builds atop these foundational Handlers in the form of `StageDefinitionBuilder` and other classes to build the DAG and implement actual Spinnaker orchestration logic.
```

Comment:
For every message *type* there is a corresponding handler? IOW, it sounds like there is a handler instance for each message rather than a handler instance waiting for any/all messages of that type.